aws/table_aws_servicecatalog_provisioned_product.go

```diff
@@ -0,0 +1,369 @@
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_servicecatalog_provisioned_product.listServiceCatalogProvisionedProducts", "api_error", err)
+			return nil, err
+		}
+
+		for _, item := range output.ProvisionedProducts {
+			d.StreamListItem(ctx, &servicecatalog.DescribeProvisionedProductOutput{
+				ProvisionedProductDetail: &types.ProvisionedProductDetail{
+					Arn:                                item.Arn,
+					CreatedTime:                        item.CreatedTime,
+					Name:                               item.Name,
+					Id:                                 item.Id,
+					IdempotencyToken:                   item.IdempotencyToken,
+					LastProvisioningRecordId:           item.LastProvisioningRecordId,
+					LastRecordId:                       item.LastRecordId,
+					LastSuccessfulProvisioningRecordId: item.LastSuccessfulProvisioningRecordId,
+					ProductId:                          item.ProductId,
+					ProvisioningArtifactId:             item.ProvisioningArtifactId,
+					Status:                             item.Status,
+					StatusMessage:                      item.StatusMessage,
+					Type:                               item.Type,
+				},
+			})
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getServiceCatalogProvisionedProduct(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var id, name string
+	id = d.EqualsQualString("id")
+	name = d.EqualsQualString("name")
+
+	if id == "" && name == "" {
+		return nil, nil
+	}
+
+	if id != "" && name != "" {
+		return nil, fmt.Errorf("Both ProvisionedProductName and ProvisionedProductId cannot be passed in the where clause simultaneously.")
+	}
+
+	// Create client
+	svc, err := ServiceCatalogClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_servicecatalog_provisioned_product.getServiceCatalogProvisionedProduct", "client_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region check
+		return nil, nil
+	}
+
+	// Input parameter
+	params := &servicecatalog.DescribeProvisionedProductInput{}
+	// Get call by passing id or name
+	if id != "" {
+		params.Id = aws.String(id)
+	} else {
+		params.Name = aws.String(name)
+	}
+	op, err := svc.DescribeProvisionedProduct(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_servicecatalog_provisioned_product.getServiceCatalogProvisionedProduct", "api_error", err)
+		return nil, err
+	}
+	return op, nil
+}
+
+//// UTILITY FUNCTIONS
+
+// Buid servicecatalog product list call filter param
+
+func buildServiceCatalogProvisionedProductFilter(ctx context.Context, quals plugin.KeyColumnQualMap) map[string][]string {
```

Comment:
@madhushreeray30, I believe this function is building the input parameter with an incorrect format. Have you tested it by providing any of the quals values in the where clause?

According to doc, `When the key is SearchQuery, the searchable fields are arn, createdTime, id, lastRecordId, idempotencyToken, name, physicalId, productId, provisioningArtifactId, type, status, tags, userArn, userArnSession, lastProvisioningRecordId, lastSuccessfulProvisioningRecordId, productName, and provisioningArtifactName.`. Means the filter parameter should look like `"SearchQuery":["status:AVAILABLE"]` not like `"status":["AVAILABLE"]` 

I have a few queries for you to review, please take a look thank!

- Is the `search_query` still necessary as an optional qualifier?
- In case it is, what would be the outcome if a user provides `search_query` along with other optional qualifiers simultaneously?
- Regarding the `created_time` column, which has a timestamp data type: are you obtaining the timestamp value by passing `string` as the dataType to the `getQualsValueByColumn(quals, columnName, "string")` function?
- Is it possible to implement support for operators in the optional qualifier `created_time`?
- Do we still need to include `id` and `name` as optional qualifiers? I am under the impression that the get config will manage them since we have utilized the `plugin.AnyColumn` function in the key columns.